quota/etcd/storage/quota_storage_test.go

```diff
@@ -393,24 +393,32 @@ func TestQuotaStorage_UpdateConfigsErrors(t *testing.T) {
 
 	want := &storagepb.Configs{} // default cfgs is empty
 	if _, err := qs.UpdateConfigs(ctx, true /* reset */, updater(want)); err != nil {
-		t.Fatalf("UpdateConfigs() returned err = %v", err)
+		t.Fatalf("UpdateConfigs()= _,%v want: _,nil", err)
 	}
 
 	for _, test := range tests {
-		if _, err := qs.UpdateConfigs(ctx, false /* reset */, test.update); !strings.Contains(err.Error(), test.wantErr) {
-			// Fatal because the config has been changed, which will break all following tests.
-			t.Fatalf("%v: UpdateConfigs() returned err = %v, want substring %q", test.desc, err, test.wantErr)
-		}
+		t.Run(test.desc, func(t *testing.T) {
+			_, err := qs.UpdateConfigs(ctx, false /* reset */, test.update)
+			// First check that an error did / did not occur as the test expects.
+			if (err != nil) != (test.wantErr != "") {
+				t.Fatalf("UpdateConfigs()=_, %v, want: _, %v", err, test.wantErr)
+			}
+			// Then if we got an error check that it was the expected one.
+			if err != nil && !strings.Contains(err.Error(), test.wantErr) {
+				// Fatal because the config has been changed, which will break all following tests.
+				t.Fatalf("UpdateConfigs()=_,%v, want substring %q", err, test.wantErr)
+			}
 
-		stored, err := qs.Configs(ctx)
-		if err != nil {
-			t.Errorf("%v:Configs() returned err = %v", test.desc, err)
-			continue
-		}
-		if got := stored; !proto.Equal(got, want) {
-			diff := pretty.Compare(got, want)
-			t.Fatalf("%v: post-Configs() diff (-got +want)\n%v", test.desc, diff)
-		}
+			stored, err := qs.Configs(ctx)
```

Comment:
nit: I wouldn't have thought we'd want to do this if UpdateConfigs() failed, but the test passes so I guess it's fine.

Reply:
My assumption was that in the error case we still want to ensure the quota state hasn't changed. There are no non-error test cases so I tweaked it slightly.

This test was pretty unclear, I suspect it isn't testing all the possible combinations and could probably be made more readable.